client/my-sites/people/edit-team-member-form/edit-user-form.jsx

```diff
@@ -39,6 +38,13 @@ const fieldKeys = {
 class EditUserForm extends Component {
 	state = this.getStateObject( this.props );
 
+	static getDerivedStateFromProps( { isExternalContributor }, state ) {
+		if ( isExternalContributor !== undefined && state.isExternalContributor === undefined ) {
+			return { isExternalContributor };
+		}
+		return null;
+	}
```

Comment:
This is a nice bugfix. One extra improvement would be to disable the "is external contributor" checkbox while its status is `undefined` i.e. unknown.

Reply:
https://github.com/Automattic/wp-calypso/pull/59946/commits/bf9023a41ff2ce25d3c1e20f528402579a20a99c

Reply:
Yes that looks like a good fix :+1: